query/src/formats/format_csv.rs

```diff
@@ -48,29 +48,46 @@ impl InputState for CsvInputState {
 pub struct CsvInputFormat {
     schema: DataSchemaRef,
     field_delimiter: u8,
-    need_skip_header: bool,
-    row_delimiter: Option<u8>,
+    skip_rows: usize,
+    record_delimiter: Option<u8>,
     min_accepted_rows: usize,
     min_accepted_bytes: usize,
     settings: FormatSettings,
 }
 
 impl CsvInputFormat {
     pub fn register(factory: &mut FormatFactory) {
-        factory.register_input(
-            "csv",
-            Box::new(
-                |name: &str, schema: DataSchemaRef, settings: FormatSettings| {
-                    CsvInputFormat::try_create(name, schema, settings, 8192, 10 * 1024 * 1024)
-                },
-            ),
-        )
+        macro_rules! register {
+            ($name: expr, $skip_rows: expr) => {
+                factory.register_input(
+                    $name,
+                    Box::new(
+                        |name: &str, schema: DataSchemaRef, settings: FormatSettings| {
+                            CsvInputFormat::try_create(
+                                name,
+                                schema,
+                                settings,
+                                $skip_rows,
+                                8192,
+                                10 * 1024 * 1024,
+                            )
+                        },
+                    ),
+                );
+            };
+        }
+
+        // TODO validate Names & Types
+        register! { "Csv", 0 }
+        register! { "CsvWithNames", 1 }
+        register! { "CsvWithNamesAndTypes", 2 }
     }
 
     pub fn try_create(
         _name: &str,
         schema: DataSchemaRef,
         settings: FormatSettings,
+        skip_rows: usize,
```

Comment:
clickhouse say in doc it support mapping by column names.
does it mean the column order of  csv may be diff from table schema?

https://clickhouse.com/docs/en/interfaces/formats#tabseparatedwithnames

of course, we can delay it.